extension/lib/catalog/product/entity/ShopgateImages.js

```diff
@@ -0,0 +1,18 @@
+/**
+ * @typedef {Object} ShopgateProductImagesDefinition
+ * @property {string[]} [images]
+ */
+
+/**
+ * @type ShopgateProductImagesDefinition
+ */
+class ShopgateImages {
+  /**
+   * @param {string[]} [images]
+   */
+  constructor (/** @type {ShopgateProductImagesDefinition} */ { images }) {
```

Comment:
not completely sure, about I think the space characters in { images } should be removed